libs/std/stdio.d

```diff
@@ -2,8 +2,17 @@
 module std.stdio;
 
 
-void writeln(string s)
-{
-    printf(s);
-    printf("\n");
+	
+void writeln(string s) {
+	// NOTE sdc cannot take the pointer from string literals.
+	// Also, string literals are null terminated by default.
+	static immutable string formatString = "%.*s\n";
```

Comment:
That is false, and there is plenty of examples in SDC's codebase that use printf("foo".ptr);